src/app/public/theming/theme-if.directive.ts

```diff
@@ -39,44 +46,44 @@ export class SkyThemeIfDirective implements OnDestroy {
   @Input()
   public set skyThemeIf(value: 'default' | 'modern') {
     this.context = value;
-    this.updateView();
-  }
-
-  private set themeSettings(settings: SkyThemeSettings) {
-    this.currentTheme = settings;
-    this.updateView();
   }
 
   private context: string;
   private currentTheme: SkyThemeSettings | undefined;
+  private embeddedView: EmbeddedViewRef<any> | undefined;
   private ngUnsubscribe = new Subject();
-  private hasView = false;
 
   constructor(
     private themeSvc: SkyThemeService,
     private templateRef: TemplateRef<any>,
     private viewContainer: ViewContainerRef
-  ) {
-    this.themeSvc.settingsChange
-      .pipe(takeUntil(this.ngUnsubscribe))
-      .subscribe(settingsChange => {
-        this.themeSettings = settingsChange.currentSettings;
-      });
+  ) {}
+
+  public ngOnChanges(): void {
+    this.updateView();
```

Comment:
What's the reasoning behind using `OnChanges`? I'd either use getters/setters _or_ use `OnChanges` (not a mix of both) if it can be avoided.

Reply:
I switched to `OnChanges` to shift the DOM changes to a lifecycle hook. I've dropped the getters/setters since they weren't needed.